### PR TITLE
[stable/elasticsearch-exporter] add ingress support (helm#22630)

### PR DESCRIPTION
**Pull request by jinleileiking**
#### What this PR does / why we need it:

Ingress support.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Review by svenmueller — COMMENTED** [review]
What is the idea behind exposing the Prometheus exporter endpoint to the public? A common recommendation would be to deploy a proxy (with auth) in front of the ES Prometheus exporter.

File: stable/elasticsearch-exporter/Chart.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 apiVersion: v1
 description: Elasticsearch stats exporter for Prometheus
 name: elasticsearch-exporter
-version: 3.3.0
+version: 3.3.1
```
Comment (caarlos0) [inline stable/elasticsearch-exporter/Chart.yaml]: shoul be `3.4.0`, as its a new feature.

File: stable/elasticsearch-exporter/README.md [hunk 2]
```diff
@@ -71,6 +71,11 @@ Parameter | Description | Default
 `service.metricsPort.name` | name for the http service | `http`
 `service.annotations` | Annotations on the http service | `{}`
 `service.labels` | Additional labels for the service definition | `{}`
+`ingress.enabled` | If true, create an ingress resource | `false`
+`ingress.annotations` | Optional ingress annotations | `{}`
+`ingress.path` | Ingress path | `/`
+`ingress.hosts` | Ingress hostnames | `kafka-manager.local`
```
Comment (caarlos0) [inline stable/elasticsearch-exporter/README.md]: why kafka?

**Review by caarlos0 — CHANGES_REQUESTED** [review]

**Comment by jinleileiking** [comment]
@caarlos0 all fixed.

**Comment by jinleileiking** [comment]
> What is the idea behind exposing the Prometheus exporter endpoint to the public? A common recommendation would be to deploy a proxy (with auth) in front of the ES Prometheus exporter.

Here is one case needs ingress:

The web ui of  prometheus -> target -> endpoint 

If we use clusterip. Then cannot show then endpoint info. But ingress can

**Comment by caarlos0** [comment]
> The web ui of prometheus -> target -> endpoint

that makes sense if your prometheus is outside the cluster the exporter is running.

Still I think @svenmueller got a point in saying you should use a proxy with auth, otherwise anyone can see your metrics... 🤔

**Comment by svenmueller** [comment]
Currently none of the standard Prometheus exporter is exposing metrics using an ingress out of the box. I don't think we should add this feature in the Elasticsearch Prometheus exporter Helm chart.

**Comment by jinleileiking** [comment]
ok close for now.

**Pull request closed without merging**